src/main/java/uk/gov/hmcts/dts/fact/controllers/SearchController.java

```diff
@@ -59,6 +64,17 @@ public SearchController(final CourtService courtService) {
         }
     }
 
+    @GetMapping(path = "/results/{postcode}")
+    @ApiOperation("Find closest courts by postcode")
+    @Description("Endpoint to return the 10 closest courts for a provided postcode")
+    @SuppressWarnings("PMD.UseObjectForClearerAPI")
+    public ResponseEntity<List<CourtReferenceWithDistance>> findCourtsByPostcode(
+        @Pattern(regexp = "^[a-z]{1,2}\\d[a-z\\d]?\\s*\\d[a-z]{2}$",
+            message = "Postcode does not match regex pattern")
```

Comment:
If this error message going to be shown externally? If so I think 'regex' sounds quite technical

Reply:
Think I agree with @KianKwa here, is there no existing string for a bad postcode anywhere?

Reply:
@Thomas-Geraghty  There is a simple one within InvalidPostcodeException class. @amstevenson is that possible to throw exception within spring boot validator so it can be caught by the exception handler, or does it need to be an explicit message?

Reply:
Explicit message @KianKwa , I have changed it to: "Provided postcode is not valid". It is important to note though that it will be the frontend querying the response, so the user shouldn't be the one seeing this :)

Reply:
Discussed with Kian and this can be resolved